pkg/operator/ceph/file/status.go

```diff
@@ -29,31 +29,41 @@ import (
 
 // updateStatus updates a fs CR with the given status
 func (r *ReconcileCephFilesystem) updateStatus(observedGeneration int64, namespacedName types.NamespacedName, status cephv1.ConditionType, info map[string]string) {
-	fs := &cephv1.CephFilesystem{}
-	err := r.client.Get(r.opManagerContext, namespacedName, fs)
-	if err != nil {
-		if kerrors.IsNotFound(err) {
-			logger.Debug("CephFilesystem resource not found. Ignoring since object must be deleted.")
-			return
+	const maxRetries = 3
+	var err error
+	retryMsg := " [retrying]"
+	for i := 0; i < maxRetries; i++ {
```

Comment:
It would be better to use the "k8s.io/client-go/util/retry" retry.RetryOnConflict() function that is part of the k8s go API.